lib/linguist/heuristics.rb

```diff
@@ -61,9 +61,6 @@ def call(data)
       @heuristic.call(data)
     end
 
-    # Common heuristics
-    ObjectiveCRegex = /^[ \t]*@(interface|class|protocol|property|end|synchronised|selector|implementation)\b/
```

Comment:
There are two different ways for detecting ObjC, each one for a different file type.

`.h` will be something like a class, so it has the check at line 81

`.m` will include something, so there's that.